e2e/ctl_v3_snapshot_test.go

```diff
@@ -119,3 +126,152 @@ func getSnapshotStatus(cx ctlCtx, fpath string) (snapshotStatus, error) {
 	}
 	return resp, nil
 }
+
+func TestCtlV3SnapshotRestore(t *testing.T) {
+	defer testutil.AfterTest(t)
+	mustEtcdctl(t)
+	os.Setenv("ETCDCTL_API", "3")
+
+	epc, err := newEtcdProcessCluster(&etcdProcessClusterConfig{
+		clusterSize:  1,
+		initialToken: "new",
+		keepDataDir:  true,
+	})
+	if err != nil {
+		t.Fatalf("could not start etcd process cluster (%v)", err)
+	}
+
+	defer func() {
+		os.Unsetenv("ETCDCTL_API")
+		if errC := epc.Close(); errC != nil {
+			t.Fatalf("error closing etcd processes (%v)", errC)
+		}
+	}()
+
+	dialTimeout := 7 * time.Second
+	prefixArgs := []string{ctlBinPath, "--endpoints", strings.Join(epc.grpcEndpoints(), ","), "--dial-timeout", dialTimeout.String()}
+
+	// write some keys
+	kvs := []kv{{"foo1", "val1"}, {"foo2", "val2"}, {"foo3", "val3"}}
+	for i := range kvs {
+		if err = spawnWithExpect(append(prefixArgs, "put", kvs[i].key, kvs[i].val), "OK"); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	fpath := filepath.Join(os.TempDir(), "test.snapshot")
+	os.RemoveAll(fpath)
+	defer os.RemoveAll(fpath)
+
+	// save snapshot
+	if err = spawnWithExpect(
+		append(prefixArgs, "snapshot", "save", fpath),
+		fmt.Sprintf("Snapshot saved at %s", fpath),
+	); err != nil {
+		t.Fatal(err)
+	}
+
+	if err = epc.backends()[0].Stop(); err != nil {
```

Comment:
we should rename backends to process or member. backends is confusing. we can do this in another pr though.